core/templates/domain/blog/BlogPostObjectFactory.ts

```diff
@@ -0,0 +1,192 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * blog post domain objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+export interface BlogPostBackendDict {
+  'id': string;
+  'author_username': string;
+  'title': string;
+  'content': string;
+  'thumbnail_filename': string;
+  'tags': string[];
+  'url_fragment': string;
+  'last_updated'?: number;
+  'published_on'?: number;
+}
+export class BlogPostData {
+  _id: string;
+  _authorUsername: string;
+  _title: string;
+  _content: string;
+  _tags: string[];
+  _thumbnailFilename: string;
+  _urlFragment: string;
+  _lastUpdated: number;
+  _publishedOn: number;
+  constructor(
+      id: string,
+      authorUsername: string,
+      title: string,
+      content: string,
+      tags: string[],
+      thumbnailFilename: string,
+      urlFragment: string,
+      lastUpdated?: number,
+      publishedOn?: number) {
+    this._id = id;
+    this._authorUsername = authorUsername;
+    this._title = title;
+    this._content = content;
+    this._tags = tags;
+    this._thumbnailFilename = thumbnailFilename;
+    this._urlFragment = urlFragment;
+    this._lastUpdated = lastUpdated;
+    this._publishedOn = publishedOn;
+  }
+
+  getId(): string {
+    return this._id;
+  }
+
+  getAuthorUsername(): string {
+    return this._authorUsername;
+  }
+
+  getLastUpdated(): number {
+    return this._lastUpdated;
+  }
+
+  getPublishedOn(): number {
+    return this._publishedOn;
+  }
+
+  setTitle(title: string): void {
+    this._title = title;
+  }
+
+  getTitle(): string {
+    return this._title;
+  }
+
+  getTags(): string[] {
+    return this._tags.slice();
+  }
+
+  addTag(tag: string): void {
+    this._tags.push(tag);
+  }
+
+  removeTag(tag: string): void {
+    let index = this._tags.indexOf(tag);
+    this._tags.splice(index, 1);
+  }
+
+  getUrlFragment(): string {
+    return this._urlFragment;
+  }
+
+  setThumbnailFilename(thumbnailFilename: string): void {
+    this._thumbnailFilename = thumbnailFilename;
+  }
+
+  getThumbnailFilename(): string {
+    return this._thumbnailFilename;
+  }
+
+  getContent(): string {
+    return this._content;
+  }
+
+  setContent(content: string): void {
+    this._content = content;
+  }
+
+  validate(): string[] {
+    var issues = [];
+    if (this._title === '') {
+      issues.push(
+        'Blog Post title should not be empty.');
+    }
+    if (this._content === '') {
+      issues.push(
+        'Blog Post content should not be empty');
+    }
+    return issues;
+  }
+
+  prepublishValidate(maxTags: number): string[] {
+    var issues = [];
+    if (this._title === '') {
+      issues.push(
+        'Blog Post title should not be empty.');
+    }
+    if (!this._thumbnailFilename) {
+      issues.push(
+        'Blog Post should have a thumbnail');
+    }
+    if (!this._tags.length) {
```

Comment:
```suggestion
    if (this._tags.length === 0) {
```

Reply:
Should we also have validation for the maxTags value to ensure that it is not zero? If the maxTags value is set to zero, it will be impossible for the blog to be published.

Reply:
Fixed, added a check in blog_admin.py to ensure that blog admin cannot set a value of 0 for max